### ISSUE TITLE
Feature request: Clock/date/battery/info on home screen

### ISSUE THREAD

**Issue by jacksongoode** [markusfisch/PieLauncher#83]
This might be out of the scope of this project or might break the philosophy of minimalism here, but i think it would be cool to have a clock and date and battery level on the home page. This would provide some basic information to folks just turning on the screen.

This could eventually be expanding to notifications or even user added widgets. But that seems like a big revamp.

The great to hear your thoughts on this approach.

**Comment by markusfisch** [comment]
Yes, basically, this would mean supporting widgets. Doing anything like widgets, even if it starts out lighter, would eventually lead to reinventing wheel 😉 

Now, the problem with having widgets is that they would interfere with opening the pie menu and, more importantly, the app drawer. Widgets can consume taps, and this would collide with opening the app drawer, and also with opening the pie menu.

So adding widgets would kind of take the best part of this launcher away, I'm afraid.

**Comment by jacksongoode** [comment]
That makes sense and a good rationale for not having anything interactive on the home screen. Would you consider adding the option for a clock or basic information that's non-interactive over the wallpaper? If not, I understand the philosophy and you can close this issue.

**Comment by markusfisch** [comment]
Hm, well, a clock would be possible, of course. Let me think about this.

Just out of curiosity, may I ask why you want a clock over the wallpaper? I mean, there's always a small clock in the status bar on the top of the screen, and probably on your lock screen too. Is it for the date?

**Comment by clshi2022** [comment]
> Just out of curiosity, may I ask why you want a clock over the wallpaper? I mean, there's always a small clock in the status bar on the top of the screen, and probably on your lock screen too. Is it for the date?

I have the same request and might be able to offer some inputs on this:
1. I have Presbyopia (getting worse), so larger text and numbers are easier to read
2. Time and icons on the status bar at the top are not always visible on certain background (e.g. light colours)
3. Some text / numbers placed appropriately often enhance the look of a visual artefact situated within borders (phone frame)
4. Subjectively having a minimal amount of utility on the minimal home screen enhances user experence

Elaborating on s/n 4, personally my main reason for having a clean homepage (i've done it on every launcher i used), is to put a nice wallpaper that makes me happy. But if i rationalised that if i put a nice wallpaper, there's no point having it covered by various icons (or too many widgets). However at the same time, this means that when the home screen has lessened utility, u don't actually need to look at it. Making it self-defeating. So the key is to balance a minimal amount of key utility that u spend just 1 second on. Arguably i would assume many people look at their phones often just to check the time or battery life. At least i personally do. Also my phone unlocks immediatly with finger print on the power button and goes straight to homepage.

**Comment by clshi2022** [comment]
**Workarounds**
Frankly there are already available workarounds and the solution is surprisingly simple. We just need to apply a live wallpaper that has a clock overlay. As shown [here](https://www.youtube.com/shorts/SojVDZGam5Q). In this video you can also see that the lighter colour at the top right corner makes any text, numbers and icons in the top status bar difficult to read compared to the black top left corner.

However the workaround is not perfect as the clock (Live Wallpaper) also shows up on the lock screen as shown below. Note that i had to remove my phone's default clock (originally a large centered digital clock)  from the lock screen otherwise they will overlap, however my phone does not have the option to remove the small one at the top left corner.

![Screenshot_20240518-021203_Pie_Launcher](https://github.com/markusfisch/PieLauncher/assets/118523010/94d0cda0-a25f-49d4-a8aa-bc4d3f1d30e8)

**Comment by markusfisch** [comment]
I see 🤔  Thanks for the explanation! Maybe I'll add a clock or something like that.
